swupd/hash.go

```diff
@@ -241,3 +173,67 @@ func (h *Hash) Sum() string {
 	hex.Encode(result[:], h.hmac.Sum(nil))
 	return string(result[:])
 }
+
+// GetHashFile calculate the swupd hash for a file in the disk.
+func GetHashFile(filename string) (string, error) {
+	var info syscall.Stat_t
+	var err error
+	if err = syscall.Lstat(filename, &info); err != nil {
+		return "", fmt.Errorf("error statting file '%s' %v", filename, err)
+	}
+
+	hashInfo := &HashFileInfo{
+		Mode: info.Mode,
+		UID:  info.Uid,
+		GID:  info.Gid,
+		Size: info.Size,
+	}
+
+	if info.Mode&syscall.S_IFMT == syscall.S_IFLNK {
+		fmt.Printf("\n\n\n%s  %x  %x  %x\n\n\n\n", filename, info.Mode, syscall.S_IFLNK, info.Mode&syscall.S_IFLNK)
```

Comment:
Debugging Printf ???

Reply:
Haha, sorry for that :-)  Fixed. Was learning how to use S_IFMT.